src/flp/types.rs

```diff
@@ -486,14 +519,20 @@ where
             )));
         }
 
-        let max = F::Integer::from(F::one());
-        for value in measurement {
-            if *value > max {
-                return Err(FlpError::Encode("Count value must be 0 or 1".to_string()));
+        let mut flattened = Vec::with_capacity(self.flattened_len);
+        for summand in measurement {
+            if summand > &self.max {
+                return Err(FlpError::Encode(format!(
+                    "summands is out of range [0, {})",
+                    1 << self.bits
```

Comment:
Just in case, we should make sure the compiler doesn't default this to i32, as that would overflow too early for some fields.
```suggestion
                    1u128 << self.bits
```

Reply:
Good catch. Nit-picking a bit, this would overflow for fields larger than 128 bits. Instead of adding a weird edge case here, I think i'll just not try to print this number.

Reply:
Oh that's a good idea :)

Reply:
Done.